plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java

```diff
@@ -10,193 +10,266 @@
-    /** Prefix for attribute of user profile that store private SSH key. */
+    // Prefix for attribute of user profile that store private SSH key.
     private static final String PRIVATE_KEY_ATTRIBUTE_PREFIX = KEY_ATTRIBUTE_PREFIX + "private.";
-    /** Prefix for attribute of user profile that store public SSH key. */
+    // Prefix for attribute of user profile that store public SSH key.
     private static final String PUBLIC_KEY_ATTRIBUTE_PREFIX  = KEY_ATTRIBUTE_PREFIX + "public.";
 
-    private final JSch          genJsch;
-    private final UserDao       userDao;
-    private final PreferenceDao preferenceDao;
+    private final JSch   genJsch;
+    private final String profileApiUrl;
 
     @Inject
-    public UserProfileSshKeyStore(PreferenceDao preferenceDao, UserDao userDao) {
+    public UserProfileSshKeyStore(@Named("api.endpoint") String apiUrl) {
         this.genJsch = new JSch();
-        this.userDao = userDao;
-        this.preferenceDao = preferenceDao;
+        this.profileApiUrl = apiUrl + "/profile";
     }
 
+    /**
+     * Adds the private key in the user preferences.
+     *
+     * @param host
+     *         host name
+     * @param key
+     *         private key as byte array
+     * @throws SshKeyStoreException
+     *         occurs when a private key already exists,
+     *         or problems with getting/updating existing keys
+     */
     @Override
     public void addPrivateKey(String host, byte[] key) throws SshKeyStoreException {
-        User user = getUser(getUserId());
         try {
             final String sshKeyAttributeName = sshKeyAttributeName(host, PRIVATE);
-            final Map<String, String> preferences = preferenceDao.getPreferences(user.getId());
-            if (preferences.get(sshKeyAttributeName) != null) {
-                throw new SshKeyStoreException(String.format("Private key for host: '%s' already exists. ", host));
+            final Map<String, String> keys = getSshKeys();
+            if (keys.get(sshKeyAttributeName) != null) {
+                throw new SshKeyStoreException("Private key for host: '" + host + "' already exists.");
             }
-            preferences.put(sshKeyAttributeName, new String(key));
-            preferenceDao.setPreferences(user.getId(), preferences);
-        } catch (NotFoundException | ServerException e) {
-            throw new SshKeyStoreException(String.format("Failed to add private key for host '%s'.", host));
+            keys.put(sshKeyAttributeName, new String(key));
+            updateSshKeys(keys);
+        } catch (ApiException | IOException | JsonParseException e) {
+            throw new SshKeyStoreException("Failed to add private key for host '" + host + "'. ", e);
         }
     }
 
+    /**
+     * Gets private key from user preferences.
+     *
+     * @param host
+     *         host name
+     * @return private key
+     * @throws SshKeyStoreException
+     *         when any error occurs with getting key
```

Comment:
AFAIK this new line does nothing. It is required <br> to create new line. Can you check?